src/test/kotlin/data/repo/ProjectRepositoryImplTest.kt

```diff
@@ -0,0 +1,342 @@
+        every { dataSource.read() } returns listOf(project)
+        justRun { dataSource.update(project.id, project) }
+
+        // When
+        val result = repo.update(project.id, project)
+
+        // Then
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `should return true when project is updated successfully`() {
+        // Given
+        val project1 = makeProject(UUID.randomUUID())
+        val project2 = makeProject(UUID.randomUUID())
+        val updatedProject = makeProject(project1.id)
+        every { dataSource.read() } returns listOf(project1, project2)
+        justRun { dataSource.update(project1.id, updatedProject) }
+
+        // When
+        val result = repo.update(project1.id, updatedProject)
+
+        // Then
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `should return false when delete a non existing project`() {
+        // Given
+        val project = makeProject(UUID.randomUUID())
+        every { dataSource.read() } returns listOf()
+
+        // When
+        val result = repo.delete(project.id)
+
+        // Then
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `should return true when delete an existing project`() {
+        // Given
+        val project = makeProject(UUID.randomUUID())
+        every { dataSource.read() } returns listOf(project)
+        justRun { dataSource.delete(project.id) }
+
+        // When
+        val result = repo.delete(project.id)
+
+        // Then
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `should return false when project not exist`() {
+        // Given
+        val project = makeProject(UUID.randomUUID())
+        every { dataSource.read() } returns listOf()
+
+        // When
+        val result = repo.exists(project.id)
+
+        // Then
+        assertThat(result).isFalse()
+    }
+
+    @Test
+    fun `should return true when project exist`() {
+        // Given
+        val project = makeProject(UUID.randomUUID())
+        every { dataSource.read() } returns listOf(project)
+
+        // When
+        val result = repo.exists(project.id)
+
+        // Then
+        assertThat(result).isTrue()
+    }
+
+    @Test
+    fun `should return project when project exist`() {
+        // Given
+        val project = makeProject(UUID.randomUUID())
+        every { dataSource.read() } returns listOf(project)
+
+        // When
+        val result = repo.get(project.id)
+
+        // Then
+        assertThat(result).isEqualTo(project)
+    }
+
+    @Test
+    fun `should throw ProjectNotFoundException when project does not exist`() {
+        // Given
+        val nonExistentId = UUID.randomUUID()
+        every { dataSource.read() } returns listOf()
+
+        // When
+        val exception = org.junit.jupiter.api.assertThrows<ProjectNotFoundException> {
+            repo.get(nonExistentId)
+        }
+
+        // Then
+        assertThat(exception.message).contains(nonExistentId.toString())
```

Comment:
The `When` should be `When & Then`, please avoid using multiple assertions in a single test.